packages/lab/src/preview.ts

```diff
@@ -0,0 +1,331 @@
+          if (iframe.contentDocument!.querySelector('.reveal')) {
+            clearInterval(waitForReveal);
+            this._ready.resolve();
+          }
+        }, 500);
+      };
+
+      this.content.url = this.getRiseUrl(this.path, index);
+      iframe.contentWindow!.addEventListener('load', setReady);
+    } else {
+      if (iframe.contentWindow) {
+        iframe.contentWindow.history.pushState(
+          null,
+          '',
+          this.getRiseUrl(this.path, index)
+        );
+      }
+    }
+  }
+
+  protected get path(): string {
+    return this._path;
+  }
+  protected set path(v: string) {
+    if (v !== this._path) {
+      this._path = v;
+      this.setActiveCellIndex(0);
+    }
+  }
+
+  protected getRiseUrl: (path: string, index?: number) => string;
+  private _ready: PromiseDelegate<void>;
+  private _renderOnSave: boolean;
+  private _path: string;
+}
+
+/**
+ * A namespace for RisePreview statics.
+ */
+export namespace RisePreview {
+  /**
+   * Instantiation options for `RisePreview`.
+   */
+  export interface IOptions
+    extends DocumentWidget.IOptionsOptionalContent<IFrame, INotebookModel> {
+    /**
+     * Application commands registry
+     */
+    commands: CommandRegistry;
+    /**
+     * The Rise URL function.
+     */
+    getRiseUrl: (path: string, index?: number) => string;
+
+    /**
+     * Whether to reload the preview on context saved.
+     */
+    renderOnSave?: boolean;
+  }
+}
+
+export class RisePreviewFactory extends ABCWidgetFactory<
+  RisePreview,
+  INotebookModel
+> {
+  defaultRenderOnSave = false;
+
+  constructor(
+    private getRiseUrl: (path: string) => string,
+    private commands: CommandRegistry,
+    options: DocumentRegistry.IWidgetFactoryOptions<RisePreview>
+  ) {
+    super(options);
+  }
+
+  protected createNewWidget(
+    context: DocumentRegistry.IContext<INotebookModel>
+  ): RisePreview {
+    return new RisePreview({
+      context,
+      commands: this.commands,
+      getRiseUrl: this.getRiseUrl,
+      renderOnSave: this.defaultRenderOnSave
+    });
+  }
+}
+
+namespace Private {
+  /**
+   * Namespace for the checkbox widget
+   */
+  export namespace CheckBox {
+    /**
+     * Constructor options for the checkbox
+     */
+    export interface IOptions {
+      /**
+       * Checkbox initial value
+       */
+      checked?: boolean;
+      /**
+       * Callback on checked status changes
+       */
+      onChange?: (ev: Event) => void;
+      /**
+       * Translator
+       */
+      translator?: ITranslator;
+    }
+  }
+
+  /**
+   * Simple checkbox
+   */
+  export class CheckBox extends Widget {
+    constructor(options: CheckBox.IOptions = {}) {
+      const trans = (options.translator ?? nullTranslator).load('rise');
+      const node = document.createElement('label');
+      node.insertAdjacentHTML(
+        'afterbegin',
+        `<input name="renderOnSave" type="checkbox"></input>${trans.__(
+          'Render on Save'
+        )}`
```

Comment:
I would suggest adding a class name + style to move the checkbox a little bit down (e.g. `top: 4px; position: relative;`) as it is currently out of line:

![Screenshot from 2022-06-21 21-07-11](https://user-images.githubusercontent.com/5832902/174889090-683d4e7c-97a0-435d-aea0-85237d56e500.png)